openedx_events/event_bus/__init__.py

```diff
@@ -1 +1,83 @@
 """Classes and utility functions for the event bus."""
+
+from abc import ABC, abstractmethod
+from typing import NoReturn, Optional
+
+from django.conf import settings
+from django.utils.module_loading import import_string
+
+
+class EventBus(ABC):
+    """
+    Parent class for event bus implementations.
+    """
+
+    @abstractmethod
+    def send(
+            *, signal: OpenEdxPublicSignal, topic: str, event_key_field: str, event_data: dict,
+            sync: bool = False,
+    ) -> None:
+        """
+        Send a signal event to the event bus under the specified topic.
+
+        Arguments:
+            signal: The original OpenEdxPublicSignal the event was sent to
+            topic: The event bus topic for the event
+            event_key_field: Path to the event data field to use as the event key (period-delimited
```

Comment:
Is this already implemented, or being introduced here? Is there an advantage to pulling this out based on some path, vs just having the caller pull it out and send the actual object they want to send?

Reply:
Already implemented in event-bus-kafka.

It's entirely possible that in many/most cases, no IDA will actually be calling this directly, and it will just be called by openedx-events based on settings. (openedx-events reads configuration, learns what signals to add receivers to, and adds receivers that convey these settings to event_bus.send.)

Reply:
I think in a previous PR or meeting we discussed having this be a function instead. Maybe we defer that discussion until we need something different?

Reply:
In https://github.com/openedx/openedx-events/issues/86 it has been suggested that OpenEdxPublicSignal could carry the topic and key-field... although I suppose we could later make it possible to override both topic and key (actual key, not key field).

Reply:
Let's defer to #86. Thanks.